### PR TITLE
Pt-1618: Quick Phenotype Search

### PR DESCRIPTION
**Pull request by Ruxuan**
Issue where Quick Phenotype Search box would overlap topmost items (PT-1618). I'd like to nominate @danielpgross  and @sdumitriu to review. Thanks!

### CODE REVIEW

**Comment by allasm** [comment]
I think it is a good idea to squash all commits into one in this case

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml [hunk 1]
```diff
@@ -107,10 +107,10 @@
         <prettyName>Use this extension</prettyName>
         <relationalStorage>0</relationalStorage>
         <separator> </separator>
-        <separators> ,|</separators>
+        <separators>|, </separators>
         <size>1</size>
         <unmodifiable>0</unmodifiable>
-        <values>currentPage=Always on this page|onDemand=On demand|always=Always on this wiki</values>
+        <values>currentPage|onDemand|always</values>
```
Comment (allasm) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: Is this change related and intentional?

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml [hunk 2]
```diff
@@ -169,13 +169,13 @@
...
-       toggleCategory(item);
-       relayoutSummaryStickyBox();
-     });
+    if (item.up('.browser') &amp;&amp; !item.up('.collapsible-group').down('input[type=checkbox]:checked')) {
+      toggleCategory(item);
+    }
+    item.observe('click', function() {
+      toggleCategory(item);
+      relayoutSummaryStickyBox();
+    });
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: These are formatting changes, not related to fixing this issue, and they should be in a separate PR.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml [hunk 3]
```diff
@@ -233,12 +233,12 @@
...
-         summaryContent.style.overflow = '';
+        summaryContent.style.height = '';
+        summaryContent.style.overflow = '';
       } else {
-         summaryContent.style.height = (summaryContent.getHeight() + (event.memo.diff || 0)) + 'px';
-         summaryContent.style.overflow = 'auto';
-         summary.style.overflow = '';
+        summaryContent.style.height = (summaryContent.getHeight() + (event.memo.diff || 0)) + 'px';
+        summaryContent.style.overflow = 'auto';
+        summary.style.overflow = '';
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: More formatting changes, same as above.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml [hunk 4]
```diff
@@ -195,7 +195,7 @@
   // ------------------------------------------------------------------------
   // Behavior of the quick search box
   // ------------------------------------------------------------------------
-  var offsetTop = $('patient-record-actions') &amp;&amp; $('patient-record-actions').getHeight() || 0;
+  var offsetTop = 0;
```
Comment (danielpgross) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: The previous value for this line added a vertical offset between the top of the viewport and the "Quick phenotype search" box, with a height equal to the patient actions toolbar. Was there a reason to remove this offset?
p0000001___phenotype_database

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml [hunk 5]
```diff
@@ -482,6 +484,10 @@
   margin: .5em .75% 0em !important;
   padding-right: .75%;
 }
+.non-moving {
+  margin-top: 75px !important;
```
Comment (danielpgross) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: `!important` should, as a rule, not be used. Is there a reason it's used here?

**Comment by danielpgross** [comment]
Thanks Ru and congrats on your first PR to the main project 👍 ! I've added my comments to the diff.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml (hunk 4 above)
Comment (Ruxuan) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: The vertical offset causes the "Quick phenotype search box" to become fixed only after scrolling past the input field. Then when it scrolls further, it jumps. Changing offsetTop to 0 causes the "Quick Phenotype search box" to become fixed at the top as soon as the view is past the top of the div making it a smoother transition. Additionally, it doesn't seem possible for the search bar and patient actions toolbar to overlap so I felt the vertical offset wasn't necessary.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml (hunk 5 above)
Comment (Ruxuan) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__phenotype.xml]: I originally had it temporarily to overwrite another !important but forgot to remove it. Good catch!
